cirq-google/cirq_google/engine/engine_client_test.py

```diff
@@ -34,6 +37,44 @@ def setup_mock_(client_constructor):
     return grpc_client
 
 
+def setup_fake_quantum_run_stream_client(client_constructor):
+    grpc_client = _FakeQuantumRunStream()
+    client_constructor.return_value = grpc_client
+    return grpc_client
+
+
+class _FakeQuantumRunStream:
+    def __init__(self):
+        self.stream_request_count = 0
+        self.cancel_requests = []
+        self.responses_and_exceptions: List[engine.QuantumRunStreamResponse | BaseException] = []
+
+    def add_responses_and_exceptions(
```

Comment:
Is this appending behavior needed, or could we make the list immutable and provide it at construction time?

Reply:
Good point, updated